### PR TITLE
Implement `--noprompt` command line argument to disable "Press any key to exit..." prompt

### PR DESCRIPTION
**Pull request by TomKranenburg**
added --noprompt command line argument to disable "Press any key to exit..." prompt

### CODE REVIEW

**Comment by TomKranenburg** [comment]
Ahhh. I think that might be notepad++ messing things up actually. It will probably be converting the line endings and it got in somehow.

I did think it was a lot of copy pasted code. But I wanted to try stick to something that was close to what you already had.

**Comment by TomKranenburg** [comment]
Ok I've added it in. It ended up being Visual Studio 2022 that was helping us to death by automatically mangling our white space and line endings. Really annoying. Just want to clarify that it wasn't Notepad++ messing up.

The code is a lot cleaner now with the callExit function I agree.

File: TinyNvidiaUpdateChecker/MainConsole.cs [hunk 1]
```diff
@@ -987,6 +936,21 @@ private static void RunIntro()
             }
         }
 
+        /// <summary>
+        /// Check for passed argument and prompt for exit if applicable
+        /// </summary>
+        /// 
+        private static void callExit(int exitnum)
```
Comment (ElPumpo) [inline TinyNvidiaUpdateChecker/MainConsole.cs:943]: Replace with `exitNum`

File: TinyNvidiaUpdateChecker/MainConsole.cs [hunk 2]
```diff
@@ -987,6 +936,21 @@ private static void RunIntro()
...
+        {
+            if (!NoPrompt)
```
Comment (ElPumpo) [inline TinyNvidiaUpdateChecker/MainConsole.cs:945]: replace `NoPrompt` with `noPrompt`

File: TinyNvidiaUpdateChecker/MainConsole.cs [hunk 3]
```diff
@@ -987,6 +936,21 @@ private static void RunIntro()
...
+            {
+                Console.WriteLine();
+                Console.WriteLine("Press any key to exit...");
+            }
```
Comment (ElPumpo) [inline TinyNvidiaUpdateChecker/MainConsole.cs:949]: whitespace between

**Review by ElPumpo — CHANGES_REQUESTED** [review]
Yea I like the impl and also your exit code argument 👍 fix these too (see review)

**Comment by TomKranenburg** [comment]
Done.

**Comment by ElPumpo** [comment]
Now finally you have not added your changes to the changelog `CHANGELOG.md`

image

Instead of `Fixed` it's `Added`

Write the following, or similar

> `--noprompt` command line argument disabling exit prompt

**Review by ElPumpo — APPROVED** [review]

**Comment by TomKranenburg** [comment]
Added @ElPumpo

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/) 
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [1.16.6] - 2023-08-20
+### Added
+- "--noprompt" command line argument disabling exit prompt (author: TomKranenburg). See issue #188
```
Comment (ElPumpo) [inline CHANGELOG.md:9]: Replace "" with ``

Also your name is no link, replace it with following:

`(author: [TomKranenburg](https://github.com/TomKranenburg)).`

**Review by ElPumpo — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (TomKranenburg) [reply CHANGELOG.md]: Done

**Comment by ElPumpo** [comment]
You're using the ` wrongly. Don't use it twice just once per side

**Comment by TomKranenburg** [comment]
Done

**Review by ElPumpo — APPROVED** [review]

**Comment by ElPumpo** [comment]
You were to use [Unreleased], not to release a new version 😄 

Thanks for your contribuations to the project!

**Pull request merged**